### PR TITLE
Add support for writing out new expected results files via env var

### PR DESCRIPTION
**Pull request by jvoravong**
- Make it so new expected test results are generate under the `functional_tests/results` directory and in the golden file format
- Use the `UPDATE_EXPECTED_RESULTS` env var to determine if new expected test results are generated, the env var should default to false.

### CODE REVIEW

File: .github/workflows/functional_test_v2.yaml [hunk 1]
```diff
@@ -62,14 +68,15 @@ jobs:
       - name: Deploy cert-manager
         run: |
           make cert-manager
-      - name: run functional tests
+      - name: Run functional tests
```
Comment (atoulme) [inline .github/workflows/functional_test_v2.yaml:71]: unrelated change?

Comment (jvoravong) [reply .github/workflows/functional_test_v2.yaml:71]: reverted

**Review by atoulme — APPROVED** [review]
LGTM

**Review by samiura — APPROVED** [review]
